core/middlewares/with-makeswift.ts

```
@@ -1,53 +1,32 @@
-import { NextRequest } from 'next/server';
-import { parse as parseSetCookie } from 'set-cookie-parser';
-
 import { routing } from '~/i18n/routing';
+import { unstable_createMakeswiftDraftRequest } from '@makeswift/runtime/next/middleware';
 
 import { MiddlewareFactory } from './compose-middlewares';
+import { strict } from 'assert';
 
 const localeCookieName = ({ localeCookie }: { localeCookie?: boolean | { name?: string } }) =>
   (typeof localeCookie === 'object' ? localeCookie.name : undefined) ?? 'NEXT_LOCALE';
 
 export const withMakeswift: MiddlewareFactory = (middleware) => {
   return async (request, event) => {
-    const apiKey =
-      request.nextUrl.searchParams.get('x-makeswift-draft-mode') ??
-      request.headers.get('x-makeswift-draft-mode');
-
-    if (apiKey === process.env.MAKESWIFT_SITE_API_KEY) {
-      const response = await fetch(new URL('/api/makeswift/draft-mode', request.nextUrl.origin), {
-        headers: {
-          'x-makeswift-api-key': apiKey,
-        },
-      });
-
-      const cookies = parseSetCookie(response.headers.get('set-cookie') || '');
-      const prerenderBypassValue = cookies.find((c) => c.name === '__prerender_bypass')?.value;
-
-      if (!prerenderBypassValue) {
-        return middleware(request, event);
-      }
-
-      // https://github.com/vercel/next.js/issues/52967#issuecomment-1644675602
-      // if we don't pass request twice, headers are stripped
-      const proxiedRequest = new NextRequest(request, request);
+    strict(process.env.MAKESWIFT_SITE_API_KEY, 'MAKESWIFT_SITE_API_KEY is required');
```

This isn't a pattern that we use in the Catalyst repo for verifying environment variables. Let's try and keep it consistent with what's already there

---

We do use `strict` in two other places for the Makeswift Site API key:

- https://github.com/bigcommerce/catalyst/blob/7dba4cc6266adb87b242b0b9ec9aac36bc20ec53/core/app/api/makeswift/%5B...makeswift%5D/route.ts#L8
- https://github.com/bigcommerce/catalyst/blob/7dba4cc6266adb87b242b0b9ec9aac36bc20ec53/core/lib/makeswift/client.ts#L10

I've gone ahead and replaced this with a top-level nullish check that throws an error for now, which is what some other scripts do for BC related env variables. I'm not really sure what the pattern for verifying environment variables is.

---

Ah, I did my global search on `canary`, not `integrations/makeswift`.

For this sort of stuff I like validating environment variables are present at build time to "fail fast" rather than having the problem show up later after deploying. Tools like https://github.com/t3-oss/t3-env are nice for this sort of stuff but there's many ways this can be done.

More generally, environment variables are a way to provide configuration to the application and ultimately configuration can come from multiple places. My approach there would be to create a configuration utility and then have that read from the necessary sources, like environment variables. Then in the application, use the configuration utility.